x-pack/plugins/uptime/public/components/monitor/__tests__/monitor_title.test.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import * as reactRouterDom from 'react-router-dom';
+import { MonitorPageTitle } from '../monitor_title';
+import { renderWithRouter, MountWithReduxProvider } from '../../../lib';
+import { AppState, store } from '../../../state';
+
+jest.mock('react-router-dom', () => ({
+  __esModule: true,
+  // @ts-ignore
+  ...jest.requireActual('react-router-dom'),
+}));
+
+describe('MonitorTitle component', () => {
+  const monitorName = 'sample monitor';
+  const monitorId = 'always-down';
+  const initialState = store.getState();
+  const url = {
+    full: 'https://www.elastic.co/',
+  };
+  const monitorStatus = {
```

Comment:
It feels like the complexity of these mocks at this point justifies a bit of refactoring here.

We do things like this elsewhere, but it's something we should avoid when we can. The reason is, if we add any additional mandatory fields to this type, this is one more file we must update. Also, it's hard to read etc. 

I am wondering if we could make a new selector in redux that just pulls the fields we need, so we would only need to mock that minimal set of fields. In general, when passing a complex mocked object, that can point to a need to reduce the surface area of the function (or component) being tested. It may accept a more complex object than it needs to. I'm also open to other approaches that could simplify the tests here.